Misc/NEWS.d/next/Core and Builtins/2024-01-11-22-58-45.gh-issue-112050.hDuvDW.rst

```diff
@@ -0,0 +1 @@
+Convert most methods on collections.deque to use Argument Clinic. This is in preparation for making deques thread-safe when the GIL is disabled. To do so, we'll use Clinic's critical_section directive.
```

Comment:
In my personal opinion, the mention of GIL and `critical_section` is not necessary here. Converting the method to utilize AC is already an independent and valuable work.

**However, I'm not ensure with this**, so if you want to keep it and other reviewers have no opinions to it, I think it's fine to keep it.

Reply:
Since this is an RST file, we should follow the style guide in the [Python Developer's Guide](https://devguide.python.org/documentation/). At least I think we should follow the line breaks, and use RST markers when referring to codes or modules.

Reply:
```suggestion
Convert :class:`collections.deque` to use Argument Clinic.
```

Reply:
We usually do not add NEWS entries for conversion to Argument Clinic. It is an internal refactoring and should not affect users.

BTW, collections.deque is not in the core.

Reply:
Despite being internal refactorings, we have, historically, provided NEWS enteies for both conversions to heap types and adaptions to Argument Clinic. I don't mind having one, but it is true it is not strictly needed either.